### PR TITLE
fix(#2568): hijack_cursor positions at top grouped node

### LINKED ISSUE #2568: Hijacked Cursor Moves To Last Grouped Folder
### Description

`renderer.group_empty` and `hijack_cursor` results in the cursor being placed at the start of the last grouped folder.

Regression since d8b154c

### Neovim version

```text
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1700008891
```

### Operating system and version

Linux 6.6.2-arch1-1

### Windows variant

_No response_

### nvim-tree version

05f55c1

### Clean room replication

```lua
vim.g.loaded_netrw = 1
vim.g.loaded_netrwPlugin = 1

vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    hijack_cursor = true,
    renderer = {
      group_empty = true,
    },
  }
end

-- UNCOMMENT this block for diagnostics issues, substituting pattern and cmd as appropriate.
-- Requires diagnostics.enable = true in setup.
--[[
vim.api.nvim_create_autocmd("FileType", {
  pattern = "lua",
  callback = function()
    vim.lsp.start { cmd = { "lua-language-server" } }
  end,
})
]]
```

### Steps to reproduce

```
mkdir -p AAAAAAAAA BBBBBBBBBB/CCCCCCCCCCCC/DDDDDDDDDDD/EEEEEEEEEEE CCCCCCCCCC DDDDDDDDDDDD/EEEEEEEE/FFFFFFFFFFFFFFFF/GGGGGGGGGGGGGGGG
touch BBBBBBBBBB/CCCCCCCCCCCC/DDDDDDDDDDD/EEEEEEEEEEE/1 DDDDDDDDDDDD/EEEEEEEE/FFFFFFFFFFFFFFFF/GGGGGGGGGGGGGGGG/2
nvim -nu /tmp/nvt-min.lua
:NvimTreeOpen
```
Open first grouped folder
Down to 1
Up one

### Expected behavior

Cursor is at start of `BBBB`

### Actual behavior

Cursor is at `EEEE`

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #2568

### CODE REVIEW

File: lua/nvim-tree/utils.lua [hunk 1]
```diff
@@ -127,10 +127,11 @@ function M.get_node_from_path(path)
     :iterate()
 end
 
--- get the highest parent of grouped nodes
-function M.get_parent_of_group(node_)
-  local node = node_
-  while node.parent and node.parent.group_next do
+---Get the highest parent of grouped nodes
+---@param node table
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:131]: Existing callers are nil safe.

File: lua/nvim-tree/utils.lua [hunk 2]
```diff
@@ -127,10 +127,11 @@ function M.get_node_from_path(path)
...
+---@return table node or parent
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:132]: Looking forward to `---@return Node node or parent`

File: lua/nvim-tree/utils.lua [hunk 3]
```diff
@@ -149,12 +149,11 @@ function M.get_node_from_path(path)
...
 end
 
---- Get the highest parent of grouped nodes
----@param node_ Node
----@return table
-function M.get_parent_of_group(node_)
-  local node = node_
-  while node.parent and node.parent.group_next do
+---Get the highest parent of grouped nodes
+---@param node Node
```
Comment (alex-courtis) [inline lua/nvim-tree/utils.lua:153]: Many thanks @Akmadan23

**Review by Akmadan23 — APPROVED** [review]
Good catch!

**Pull request merged**